content/articles/mock-a-datasource-for-react-using-json-server/index.md

```diff
@@ -0,0 +1,837 @@
+
+### Pre-requisites
+
+For the article, the following are required for easy article follow-up:
+
+- A good React IDE or text editor set up on the machine
+- A stable internet application
+- React development environment already setup
+- React development skills
+
+Brief info on what is going to be done in the article is as follows:
+
+- Learn briefs on the JSON Server package
+- Create a React application
+- Set it up to allow API data from links
+- Style the application
+- Install JSON Server in the machine
+- Set up the JSON Server as a mock database server
+- Configure the JSON Server data source in the project
+- Run the application
+
+### Learn briefs on the JSON Server package
+
+JSON Server can allow almost all back-end requests and responses. They can be accessed via the GET, POST, PUT, PATCH, and DELETE methods.
+It allows routes to access data items stored in the Mock database file(in JSON format). They may include examples such as `GET    /posts`, `PUT    /posts/1`, or `DELETE /posts/1`.
+
+The module allows other operations to be done on the database, such as:
+
+- **Filter**: e.g. `GET /posts?title=json-server&author=riro` or `GET /comments?author.name=riro`
+- **Pagination**: e.g. `GET /posts?_page=9&_limit=23`
+- **Sorting**: e.g. `GET /posts/5/comments?_sort=votes,likes&_order=desc,asc`
+- **Slicing**: e.g. `GET /posts/4/comments?_start=20&_limit=10`
+- **Operators**: e.g. for getting range (`_gte`, `lte`), excluding a value(`_ne`), filtering a value(`_like`)
+- **Full-text search**: e.g. `GET /posts?q=tomcat`
+- **Relationships**: e.g., the inclusion of child resources(`_embed`), the inclusion of parent resources(`_expand`), to get or create nested resources
+- **Database**: e.g. `GET \db`
+- **Homepage**: e.g. `GET /`. Using the ' ./public ' path, one can serve the application files in the public folder.
+
+More on JSON Server library will also be mentioned in the article.
+
+### Create a React app
+
+In this section, a React app will be created. The app will be a budget planning application to add items and remove them from a list.
+The app saves and retrieves data from a data source in a JSON object.
+
+#### New React app
+
+- Open the folder in which the application should be created and run this command:
+
+```shell
+npx create-react-app react-budget-tracker
+```
+
+#### New project structure
+- Create the following application folder structure:
+
+```shell
+.
+├── node_modules
+├── public
+├── src
+│   └── components
+│       ├── About.js
+│       ├── AddBudget.js
+│       ├── Budget.js
+│       ├── Budgets.js
+│       ├── Button.js
+│       ├── Footer.js
+│       └── Header.js
+│   ├── App.js
+│   ├── index.css
+│   ├── index.js
+│   └── reportWebVitals.js
+├── .eslintcache
+├── .gitignore
+├── db.json
+├── package.json
+├── package-lock.json
+├── README.md
+└── yarn.lock
+```
+
+#### Header component
+- In the 'Header.js', copy and paste the code below:
```

Comment:
`copy and paste`

Avoid using this.

Instead, explain and walk the reader through the process.